### PR TITLE
Clicking on the community in the sidebar must redirect the user to community home page.

### PR DESCRIPTION
**Pull request by srikar0896**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge**
- hyperion (frontend)

**Release notes for users**
- User can visit the community home by clicking the community in the sidebar.

**Related issues**
Closes https://github.com/withspectrum/spectrum/issues/4556

### CODE REVIEW

**Comment by mxstbr** [comment]
Deployed as https://spectrum-ejrmdorrgw.now.sh for testing, let's see...

File: src/views/dashboard/components/communityList.js [hunk 1]
```diff
@@ -107,36 +107,40 @@ class CommunityList extends React.Component<Props> {
...
-                {c.id === activeCommunity && (
-                  <ErrorBoundary>
-                    <SidebarChannels
-                      activeChannel={activeChannel}
-                      communitySlug={c.slug}
-                      permissions={c.communityPermissions}
-                      slug={c.slug}
-                      id={c.id}
-                      setActiveChannelObject={this.props.setActiveChannelObject}
+              <Link to={`/${c.slug}`}>
```
Comment (mxstbr) [inline src/views/dashboard/components/communityList.js]: We do not want to redirect to the community once the JS has rehydrated. I think to fix that, we need to move the `onClick` to the `Link` here!

```suggestion
              <Link to={`/${c.slug}`} onClick={() => this.handleOnClick(c.id)}>
```

File: src/views/dashboard/components/communityList.js [hunk 2]
```diff
@@ -107,36 +107,40 @@ class CommunityList extends React.Component<Props> {
...
+                <CommunityListItem
+                  onClick={() => this.handleOnClick(c.id)}
```
Comment (mxstbr) [inline src/views/dashboard/components/communityList.js]: And remove the onClick here:

```suggestion

```

**Review by mxstbr — CHANGES_REQUESTED** [review]

File: src/views/dashboard/components/communityList.js (hunk 1 above)
Comment (srikar0896) [reply src/views/dashboard/components/communityList.js]: oh yes. And adding `e.preventDefault` in `onClick` handler might satisfy that condition.

Thanks.

**Comment by mxstbr** [comment]
Hmm, this still redirects me to `/${communitySlug}` when clicking on a filter, even when the JS is running?

**Comment by srikar0896** [comment]
@mxstbr yes, the event in `sideBarChannel` is stopped from propagating to the parent but the default event for a link is to redirect to that link, so I have fixed that by adding `e.preventDeault()` because we don't want the default action tobe taken.

**Comment by srikar0896** [comment]
@mxstbr Hi!! Have you got a chance to check this?

**Comment by Ponjimon** [comment]
I had a look at this and unfortunately, it does not work for me. I mean, it does, but I'm not sure if that's intended. If I click the community item, it just redirects me to the community instead of expanding the channel list of that community as it currently does.

If you put the `e.preventDefault()` at the beginning of the onClick function to capture the regular behavior of a link, then it will work as it should, in my opinion. The current behavior is kept but additionally, it's not a link that you can open in a new tab. No need for the `else` here, we can always prevent the default.

**Pull request closed without merging**